dowhy/causal_prediction/algorithms/regularization.py

```diff
@@ -0,0 +1,300 @@
+import numpy as np
+import torch
+from torch import nn
+from torch.nn import functional as F
+
+from dowhy.causal_prediction.algorithms.utils import mmd_compute
+
+
+class Regularization:
```

Comment:
would Regularizer be a better name since class names are usually about the object (e.g., PredictionAlgorithm and not just Prediction?)